### PR TITLE
fix(lib-content): focus the contact form when it is opened

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Add a timeout to the About page, which waits for the popup contact form to open, finds the Name input and then focuses it.
- Add a keydown handler to close the popup with Esc.
- [Security] Add an integrity check to the Freshworks script.

https://github.com/user-attachments/assets/3bbba1b9-501f-42aa-9a32-05bcf30268c9

_Please request review from `@zooniverse/frontend` team or an individual member of that team._ 

## Package
- lib-content

## Linked Issue and/or Talk Post
- the popup contact form isn't accessible with the keyboard or a screenreader ([WCAG 2.1.1: Keyboard](https://www.w3.org/WAI/WCAG22/Understanding/keyboard).)

## How to Review
You should now be able to fill out the contact form with a screenreader, like NVDA or VoiceOver, or while navigating the About page with the Tab key.

Here's a video I recorded where I fill out the form in VoiceOver on this branch. Weirdly, I'm not able to fill out the email field with my stored email address, though I wouldn't know that if I couldn't see the form. I think that's a problem in Safari, with or without VoiceOver.

https://github.com/user-attachments/assets/20c3345d-a285-4b29-a754-3a9e5fca898b

This doesn't fully satisfy WCAG 2.1.1, which is a level A requirement, as the popup form still can't be closed from the keyboard. There's also no announcement, in a screenreader, that a dialog has opened, because the [ARIA dialog roles and states](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/) are missing from the popup. Likewise, the Contact Us button has no [`aria-controls`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls) or [`aria-haspopup`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup) state to indicate, to a blind reader, that it will open a popup.

EDIT: I've added `aria-haspopup=dialog` to the button, so it should at least announce that it opens a popup. I've also added a keydown handler to close the popup when Esc is pressed.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
- [ ] The PR creator has listed user actions to use when testing if bug is fixed
- [ ] The bug is fixed
- [ ] Unit tests are added or updated

### CODE REVIEW

**Comment by goplayoutside3** [comment]
> Likewise, the Contact Us button has no [aria-controls](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls) or [aria-haspopup](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-haspopup) state to indicate, to a blind reader, that it will open a popup.

I noticed you added `aria-haspopup` to the button, are there further changes you'd like to include in this PR?

**Comment by eatyourgreens** [comment]
>I noticed you added aria-haspopup to the button, are there further changes you'd like to include in this PR?

I've added a listener that closes the popup with the Esc key. I'm not sure the Freshworks widget would pass a WCAG audit, but this should help make it work with a keyboard/screenreader. I think that's about all I can do here, in terms of hacking the contact form to add keyboard support.

I tried adding [an integrity hash](https://www.srihash.org/) to the script tag that loads the freshworks script and got this error in the console (😳). The script also stopped working, since it can't be verified.
>"https://widget.freshworks.com/widgets/44000004375.js" is not eligible for integrity checks since it's neither CORS-enabled nor same-origin.

So security seems dubious for scripts loaded from `widget.freshworks.com`. An `integrity` attribute would protect against supply chain attacks like the `polyfill.io` malware (if someone were to somehow modify the freshworks script.)

File: packages/lib-content/src/screens/About/About.js [hunk 1]
```diff
@@ -70,6 +70,8 @@ function AboutPage() {
         src='https://widget.freshworks.com/widgets/44000004375.js'
         async
         defer
+        crossOrigin='anonymous'
+        integrity='sha384-GpJCLE7VNj6jp2jskwQ5wnbyfRLB4jsiVzyCg1pJuMBnAqjylfzWNhQu3yKQeidZ'
```
Comment (eatyourgreens) [inline packages/lib-content/src/screens/About/About.js:74]: This basically verifies that the content of the URL `https://widget.freshworks.com/widgets/44000004375.js` hasn't been modified from the version that was used to generate the integrity hash. If the script doesn't match, it won't run.

File: packages/lib-content/src/screens/About/components/Contact.js [hunk 2]
```diff
@@ -17,7 +17,19 @@ export default function Contact({ widgetLoaded = false }) {
...
       window.FreshworksWidget('open')
+      setTimeout(() => {
+        const iframe = document.querySelector('iframe#widget-frame')
+        const input = iframe?.contentDocument?.querySelector('input#name')
+        input?.focus()
+        iframe?.contentDocument?.addEventListener('keydown', (event) => {
+          if (event.key === 'Escape') {
+            contactButton.focus()
+            window.FreshworksWidget('close')
+          }
```
Comment (eatyourgreens) [inline packages/lib-content/src/screens/About/components/Contact.js:30]: Focus the contact button, then close the form, otherwise focus jumps back to the `<body>` when the form closes (default browser behaviour when you remove the current `document.activeElement` from the DOM.)

**Review by goplayoutside3 — COMMENTED** [review]
When I run lib-content's Storybook (`http://localhost:6005/?path=/story/about-about--default`), the Contact form opens and closes as expected, but when app-root is run locally and I visit https://local.zooniverse.org:3000/about#contact, the form will not open. The error in the console is 
```
Access to script at 'https://widget.freshworks.com/widgets/44000004375.js' from origin 'https://local.zooniverse.org:3000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

Why is that?

**Comment by eatyourgreens** [comment]
@goplayoutside3 I'm not sure. I can see an `Access-Control-Allow-Origin` header in the response from `widget.freshworks.com`, though the response is a redirect rather than the script itself.

No problems here with the contact form, in either Chrome or Firefox. This video is Chrome with Next.js running in dev mode (production mode works too.)


https://github.com/user-attachments/assets/ccc97277-a51f-435d-93b5-0bbb29c6c0b6


https://github.com/user-attachments/assets/ccc97277-a51f-435d-93b5-0bbb29c6c0b6

**Comment by eatyourgreens** [comment]
> I have a feeling it's because I've enabled ad blockers and disabled other tracking and third-party stuff for those two.

Mathias Schäfer just published [a blog post about this problem](https://molily.de/something-went-wrong/). It's a good read.

**Comment by goplayoutside3** [comment]
Seeing this same error on fe-root.preview :( 
```
Access to script at 'https://widget.freshworks.com/widgets/44000004375.js' from origin 'https://fe-root.preview.zooniverse.org' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
```

File: packages/lib-content/src/screens/About/About.js [hunk 3]
```diff
@@ -70,6 +70,8 @@ function AboutPage() {
         src='https://widget.freshworks.com/widgets/44000004375.js'
         async
         defer
+        crossOrigin='anonymous'
+        integrity='sha384-GpJCLE7VNj6jp2jskwQ5wnbyfRLB4jsiVzyCg1pJuMBnAqjylfzWNhQu3yKQeidZ' // Generated by https://www.srihash.org
```
Comment (goplayoutside3) [inline packages/lib-content/src/screens/About/About.js:74]: Unfortunately had to revert this change in https://github.com/zooniverse/front-end-monorepo/pull/6306 because zoo team members tested https://fe-root.preview.zooniverse.org/about#contact on multiple different browsers and devices, and could not open the Contact Us widget.

Comment (eatyourgreens) [reply packages/lib-content/src/screens/About/About.js:74]: Unfortunately, that does leave site users open to having their personal information hacked via `widget.freshworks.com`.

I had another look in Chrome, but the request is sending the `Origin` header, as expected, and `widget.freshworks.com` is responding with `Access-Control-Allow-Origin: *`, as expected. So I don't see anything that would block the script from loading with CORS enabled.

Request headers, including Origin, for the Freshworks JS widget.

Response headers for the Freshworks JS widget, including Access-Control-Allow-Origin
